pkg/proxy/dialer.go

```diff
@@ -65,5 +80,10 @@ func ciliumDialer(identity int, network, address string) (net.Conn, error) {
 		return nil, fmt.Errorf("unable to connect: %s", err)
 	}
 
+	if err := setKeepAlive(c); err != nil {
+		c.Close()
```

Comment:
Would it make more sense to add `defer c.Close()` on line and 61 instead? and cleanup all other `c.Close` too

Reply:
We only want to close when the operation failed

Reply:
got it ..thanks!